File: docs/exemple/jpa/src/main/javagen/topmodel/jpa/sample/demo/api/server/securite/utilisateur/UtilisateurApiController.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.api.server.securite.utilisateur;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PreAuthorize;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import jakarta.annotation.Generated;
import jakarta.validation.Valid;

import topmodel.jpa.sample.demo.dtos.utilisateur.UtilisateurDto;
import topmodel.jpa.sample.demo.entities.utilisateur.TypeUtilisateur;

@RequestMapping("utilisateur")
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface UtilisateurApiController {


	/**
	 * Charge le détail d'un utilisateur.
	 * @param utilisateurId Id technique
	 * @return Le détail de l'utilisateur
	 */
	@GetMapping(path = "/{utiId}")
	UtilisateurDto find(@RequestParam(value = "utilisateurId", required = true) Long utilisateurId);

	/**
	 * Charge une liste d'utilisateurs par leur type.
	 * @param typeUtilisateurCode Type d'utilisateur en Many to one
	 * @return Liste des utilisateurs
	 */
	@GetMapping(path = "/list")
	List<UtilisateurDto> findAllByType(@RequestParam(value = "typeUtilisateurCode", required = false) TypeUtilisateur.Values typeUtilisateurCode);

	/**
	 * Sauvegarde un utilisateur.
	 * @param utilisateur Utilisateur à sauvegarder
	 * @return Utilisateur sauvegardé
	 */
	@PreAuthorize("hasRole('ROLE_ADMIN')")
	@PostMapping(path = "/save")
	UtilisateurDto save(@RequestBody @Valid UtilisateurDto utilisateur);

	/**
	 * Recherche des utilisateurs.
	 * @param utilisateurId Id technique
	 * @param age Age en années de l'utilisateur
	 * @param profilId Profil de l'utilisateur
	 * @param email Email de l'utilisateur
	 * @param typeUtilisateurCode Type d'utilisateur en Many to one
	 * @param dateCreation Date de création de l'utilisateur
	 * @param dateModification Date de modification de l'utilisateur
	 * @return Utilisateurs matchant les critères
	 */
	@PostMapping(path = "/search")
	Page<UtilisateurDto> search(@RequestParam(value = "utilisateurId", required = true) Long utilisateurId, @RequestParam(value = "age", required = false) Long age, @RequestParam(value = "profilId", required = false) Long profilId, @RequestParam(value = "email", required = false) String email, @RequestParam(value = "typeUtilisateurCode", required = false) TypeUtilisateur.Values typeUtilisateurCode, @RequestParam(value = "dateCreation", required = false) LocalDate dateCreation, @RequestParam(value = "dateModification", required = false) LocalDateTime dateModification);
}

File: samples/generators/csharp/src/CSharp.Api/Controllers/Securite/Utilisateur/UtilisateurApiController.cs
////
//// ATTENTION, CE FICHIER EST PARTIELLEMENT GENERE AUTOMATIQUEMENT !
////

using Microsoft.AspNetCore.Mvc;

namespace CSharp.Api.Securite.Utilisateur;

public class UtilisateurApiController : Controller
{

    /// <summary>
    /// Recherche des utilisateurs
    /// </summary>
    /// <param name="utiId">Id technique</param>
    /// <returns>Task.</returns>
    [HttpDelete("utilisateur/deleteAll")]
    public async Task DeleteAll(int[] utiId = null)
    {

    }

    /// <summary>
    /// Charge le détail d'un utilisateur
    /// </summary>
    /// <param name="utiId">Id technique</param>
    /// <returns>Le détail de l'utilisateur</returns>
    [HttpGet("utilisateur/{utiId:int}")]
    public async Task<UtilisateurDto> Find(int utiId)
    {

    }

    /// <summary>
    /// Charge une liste d'utilisateurs par leur type
    /// </summary>
    /// <param name="typeUtilisateurCode">Type d'utilisateur en Many to one</param>
    /// <returns>Liste des utilisateurs</returns>
    [HttpGet("utilisateur/list")]
    public async Task<IEnumerable<UtilisateurSearch>> FindAllByType(TypeUtilisateur.Codes typeUtilisateurCode = TypeUtilisateur.Codes.ADM)
    {

    }

    /// <summary>
    /// Sauvegarde un utilisateur
    /// </summary>
    /// <param name="utilisateur">Utilisateur à sauvegarder</param>
    /// <returns>Utilisateur sauvegardé</returns>
    [HttpPost("utilisateur/save")]
    public async Task<UtilisateurDto> Save([FromBody] UtilisateurDto utilisateur)
    {

    }

    /// <summary>
    /// Recherche des utilisateurs
    /// </summary>
    /// <param name="utiId">Id technique</param>
    /// <param name="age">Age en années de l'utilisateur</param>
    /// <param name="profilId">Profil de l'utilisateur</param>
    /// <param name="email">Email de l'utilisateur</param>
    /// <param name="nom">Nom de l'utilisateur</param>
    /// <param name="actif">Si l'utilisateur est actif</param>
    /// <param name="typeUtilisateurCode">Type d'utilisateur en Many to one</param>
    /// <param name="utilisateursEnfant">Utilisateur enfants</param>
    /// <param name="dateCreation">Date de création de l'utilisateur</param>
    /// <param name="dateModification">Date de modification de l'utilisateur</param>
    /// <returns>Utilisateurs matchant les critères</returns>
    [HttpPost("utilisateur/search")]
    public async Task<ICollection<UtilisateurSearch>> Search(int? utiId = null, decimal age = 6l, int? profilId = null, string email = null, string nom = "Jabx", bool? actif = null, TypeUtilisateur.Codes typeUtilisateurCode = TypeUtilisateur.Codes.ADM, int[] utilisateursEnfant = null, DateOnly? dateCreation = null, DateOnly? dateModification = null)
    {

    }

}